### PR TITLE
gh-100176: Tools/iobench: Remove redundant compat code for Python <= 3.2

### PR DESCRIPTION
**Pull request by hugovk**
Python 3.2 has been EOL since 2016-02-20 and 2.7 since 2020-01-01, so we can remove these old compatibility checks.

https://devguide.python.org/versions/

Also use f-strings and adjust some whitespace.

---

Ping also @pitrou who created this file with this note in 2010 in https://github.com/python/cpython/commit/a69ba65fdce3be6f55634b47cc56cf1f962f6cbc:

> This file should be kept compatible with both Python 2.6 and Python >= 3.0.

---

* Issue: gh-100176

### CODE REVIEW

**Comment by hugovk** [comment]
See https://discuss.python.org/t/do-we-still-need-tools-io-cc-string-bench/22557 for a discussion about `Tools/{io,cc,string}bench`, whether to leave, modernise, move or delete them.

File: Tools/iobench/iobench.py [hunk 1]
```diff
@@ -266,18 +280,17 @@ def run_all_tests(options):
...
 
     def print_results(size, n, real, cpu):
         bw = n * float(size) / 1024 ** 2 / real
         bw = ("%4d MiB/s" if bw > 100 else "%.3g MiB/s") % bw
         out.write(bw.rjust(12) + "\n")
         if cpu < 0.90 * real:
-            out.write("   warning: test above used only %d%% CPU, "
-                "result may be flawed!\n" % (100.0 * cpu / real))
+            out.write("   warning: test above used only "
+                      f"{100.0 * cpu / real}% CPU, "
```
Comment (AlexWaygood) [inline Tools/iobench/iobench.py]: Little-known fact: Python has a builtin `%` format specifier 

```suggestion
                      f"{cpu / real:%} CPU, "
```

Comment (hugovk) [reply Tools/iobench/iobench.py]: Slightly different output but doesn't really matter here:

```pycon
>>> f"{100.0 * 12 / 36}% CPU"
'33.333333333333336% CPU'
>>> f"{12 / 36:%} CPU"
'33.333333% CPU'
```

**Review by AlexWaygood — APPROVED** [review]
LG

**Comment by hugovk** [comment]
Let's hold off a bit merging this, we can remove the file instead:

* https://github.com/python/cpython/issues/100176#issuecomment-1374752981
* https://discuss.python.org/t/do-we-still-need-tools-io-cc-string-bench/22557

**Comment by hugovk** [comment]
> Let's hold off a bit merging this, we can remove the file instead:
> 
> * [Meta issue: clean up redundant compat code #100176 (comment)](https://github.com/python/cpython/issues/100176#issuecomment-1374752981)
> * [discuss.python.org/t/do-we-still-need-tools-io-cc-string-bench/22557](https://discuss.python.org/t/do-we-still-need-tools-io-cc-string-bench/22557)
> * [gh-100176: Remove outdated Tools/{io,cc,string}bench #101853](https://github.com/python/cpython/pull/101853)

No decision there in a couple of months, let's merge this. We can still remove the file later.

**Pull request merged**